### ISSUE TITLE
The Early Exit sniff corrupts conditionally declared global symbols

### ISSUE THREAD

**Issue by morozov** [slevomat/coding-standard#1122]
Consider the following code example:
```php
<?php

if (rand(0, 1)) {
    function test(): void
    {
        echo 1;
    }

    return;
} else {
    function test(): void
    {
        echo 2;
    }
}
```
Fixing the violations in the file produces the following:
```
$ phpcbf test.php
F 1 / 1 (100%)

PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
test.php                                              1      2
----------------------------------------------------------------------
A TOTAL OF 1 ERROR WERE FIXED IN 1 FILE
----------------------------------------------------------------------
```
```php
<?php

if (rand(0, 1)) {
    function test(): void
    {
        echo 1;
    }

    return;
}

function test(): void
{
    echo 2;
}
```
```
$ php test.php

Fatal error: Cannot redeclare test() (previously declared in test.php:14) in test.php on line 4

Call Stack:
    0.0004     465504   1. {main}() test.php:0
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/23de77f26757fe622c1b6b33285c0f58acf13a95
